lib/ansible/plugins/callback/default.py

```diff
@@ -40,6 +40,10 @@ def __init__(self):
         self._last_task_banner = None
         super(CallbackModule, self).__init__()
 
+    def _dump_results(self, result, **kwargs):
+        result = { key: result[key] for key in result if key not in ['deprecation', 'warnings'] }
+        return super(CallbackModule, self)._dump_results(result, **kwargs)
+
```

Comment:
I just pushed key deletion to handle warnings, now we just need to make sure this happens before dump